src/Downloader.ts

```diff
@@ -585,20 +616,29 @@ class Downloader {
       });
   }
 
-  private async getCompressedBody(resp: any): Promise<any> {
-    return this.isMimeTypeImage(resp.headers['content-type']) ? await imagemin.buffer(resp.data, imageminOptions) : resp.data;
+  private async getCompressedBody(resp: any, path: string): Promise<any> {
+    resp.headers.path = path;
+    if (isImageMimeType(resp.headers['content-type'])) {
+      resp.data = await imagemin.buffer(resp.data, imageminOptions.get(resp.headers['content-type']));
+      resp.headers.path += '.webp';
+      resp.headers['content-type'] = 'image/webp';
+    }
+    return resp;
   }
 
-  private getContentCb = async (url: string, handler: any): Promise<void> => {
+  private getContentCb = async (args: string[], handler: any): Promise<void> => {
+    const url = args[0];
+    const path = args[1];
     logger.info(`Downloading [${url}]`);
     try {
-      if (this.optimisationCacheUrl && this.isImageUrl(url)) {
-        this.downloadImage(url, handler);
+      if (this.optimisationCacheUrl && isImageUrl(url)) {
+        this.downloadImage(url, handler, path);
       } else {
-        const resp = await axios(url, this.arrayBufferRequestOptions);
+        let resp = await axios(url, this.arrayBufferRequestOptions);
+        resp = await this.getCompressedBody(resp, path)
```

Comment:
You should not have to give the `path` through all this methods. This should be avoided.

Deal with `resp.headers.path_postfix`.